src/Lean/Elab/Structure.lean

```diff
@@ -387,8 +387,8 @@ where
         let arg ← mkFreshExprMVar d
         go? (b.instantiate1 arg)
     | e =>
-      let _ := if e.isAppOfArity ``id 2 then e.appArg! else e
-      return some (← reduceProjs (← instantiateMVars e.appArg!) expandedStructNames)
+      let r := if e.isAppOfArity ``id 2 then e.appArg! else e
+      return some (← reduceProjs (← instantiateMVars r.appArg!) expandedStructNames)
```

Comment:
@leodemoura

Reply:
Apparently not, the change breaks run/diamond{3,4}.lean

Reply:
Thanks for catching this. I pushed a fix for this.
```lean
      let r := if e.isAppOfArity ``id 2 then e.appArg! else e
      return some (← reduceProjs (← instantiateMVars r) expandedStructNames)
```